### PR TITLE
Make AmMemberOf to use only the mspIDs in collection policy

### PR DESCRIPTION
**Pull request by cendhu**
#### Type of change

- Bug fix

#### Description

https://github.com/hyperledger/fabric/issues/4229

### CODE REVIEW

File: core/common/privdata/membershipinfo.go [hunk 1]
```diff
@@ -35,33 +35,19 @@ func NewMembershipInfoProvider(mspID string, selfSignedData protoutil.SignedData
...
-
-	// Fall back to default access policy evaluation otherwise
-	accessPolicy, err := getPolicy(collectionPolicyConfig, deserializer)
-	if err != nil {
-		// drop the error and return false - same as reject all policy
-		logger.Errorf("Reject all due to error getting policy: %s", err)
-		return false, nil
-	}
-	if err := accessPolicy.EvaluateSignedData([]*protoutil.SignedData{&m.selfSignedData}); err != nil {
+	if collectionPolicyConfig.GetSignaturePolicy() == nil {
```
Comment (yacovm) [inline core/common/privdata/membershipinfo.go:43]: maybe print in warning that this is nil?

**Review by denyeart — DISMISSED** [review]
Fix looks reasonable given we never added policy support for collections.

With the fix, I am not able to reproduce in local environment (basically the same test as the new integration test).

**Review by denyeart — APPROVED** [review]

**Comment by denyeart** [comment]
@Mergifyio backport release-2.5

**Comment by denyeart** [comment]
https://github.com/Mergifyio backport release-2.2

**Pull request merged**